enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Baremetal IPI Config Enhancement
+authors:
+  - "@sadasu"
+reviewers:
+  - "@deads2k"
+  - "@hardys"
+  - "@enxebre"
+  - "@squeed"
+  - "@abhinavdahiya"
+  - "@dhellmann"
+
+approvers:
+  - "@abhinavdahiya"
+  - "@enxebre"
+  - "@deads2k"
+
+creation-date: 2019-11-14
+last-updated: 2019-11-26
+status: not implemented
+see-also:
+  - 
+replaces:
+  - https://github.com/openshift/enhancements/pull/90
+superseded-by:
+  - 
+---
+
+# Config required for Baremetal IPI deployments
+
+## Release Signoff Checklist
+
+- [*] Enhancement is `implementable`
+- [*] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions
+
+1. Which is a preferred name for the new CR "Metal3ProvisioningController" or
+"Metal3Controller"?
+
+[Closed] Based on review comments it appears that "Metal3ProvisioningController"
+is the preferred name since it makes the contents of the config very clear.
+
+2. There is a possibility that the MCO would also have to refer to this new CR
+created in operator scope. Would it be OK for the MCO to access the CR with this
+(limited) scope?
+
+[Closed] MCO will be able to read config from this new CR if needed.
+
+## Summary
+
+The configuration representing different provisioning network parameters used
+for provisioning baremetal servers need to be made available to the Machine
+API Operator (MAO) in a Config Resource. We had submitted an earlier proposal
+[1] to augment the BareMetalInfrastructureStatus CR with these config items.
+Based on feedback we recieved we are revising our proposal to create a new CR
+for these configuration items.
+
+This new CR needs to be accessed by just the Installer and the Machine API
```

Comment:
can we be specific here, the installer would be responsible for instantiating the CR with user input values whereas the mao will need to fetch it because...? (see my comment above)

Reply:
The MAO is responsible for deploying the metal3 cluster. And the configuration values from the CR would be read by the MAO and pass these values as env vars to the containers that are part of the metal3 cluster before starting the containers. In most cases, the containers will not start without these init config available as env vars. This was explained earlier in the doc but I had not mentioned what would happen if not present. Will add that too.